ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -100,46 +107,41 @@ function check_history() {
     fi
 }
 
-if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
     export KUBECONFIG=${SHARED_DIR}/kubeconfig
 fi
 
 # Setup proxy if it's present in the shared dir
-if test -f "${SHARED_DIR}/proxy-conf.sh"
-then
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
     # shellcheck disable=SC1091
     source "${SHARED_DIR}/proxy-conf.sh"
 fi
 
-echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
-echo "RELEASE_IMAGE_TARGET is ${RELEASE_IMAGE_TARGET}"
-
-echo -e "Current cluster version: oc get clusterversion\n"
-oc get clusterversion
+get_target_release
+for target in "${TARGET_RELEASES[@]}"
+do
+    echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
+    echo "RELEASE_IMAGE_TARGET is ${RELEASE_IMAGE_TARGET}"
 
-echo -e "RELEASE_IMAGE_LATEST release info:\n"
-oc adm release info "${RELEASE_IMAGE_LATEST}"
+    echo -e "Current client/server version: oc version\n"
+    oc version
 
-echo -e "RELEASE_IMAGE_TARGET release info:\n"
-oc adm release info "${RELEASE_IMAGE_TARGET}"
+    source_minor_version="$(oc get clusterversion --no-headers | awk '{print $2}' | cut -f2 -d.)"
```

Comment:
if vars (e.g: `source_minor_version`) will be referenced in the following function, I suggest always `export` it for safety.

Reply:
Thanks for the suggestion. Wrt the best practice, i'd like to seek your expertise
1) if a var is readonly to the functions and subprocesses, making it an env var better?
2) if a var is writable to all the functions, making it a global var better (by default it's a global var)?

Reply:
Yeah, you are right. Subprocesses can not write a env inherited from its parent process.